api/src/websocket/websocket.gateway.ts

```diff
@@ -405,4 +422,45 @@ export class WebsocketGateway
     );
     return response.getPromise();
   }
+
+  /**
+   * Retrieves notification sockets based on session id.
+   *
+   * @param sessionId - The session id
+   * @returns An RemoteSocket array
+   */
+  async getNotificationSockets(
+    sessionId: string,
+  ): Promise<RemoteSocket<DefaultEventsMap, any>[]> {
+    if (!sessionId) {
+      throw new Error('SessionId is required!');
+    }
+
+    return (await this.io.fetchSockets()).filter(
+      ({ handshake, data }) =>
+        !handshake.query.channel && data.sessionID === sessionId,
```

Comment:
The channel needs to be "console", right ?

Reply:
From my understanding this code will make sockets without channel query param (global) joining message and subscriber rooms to be able to see global events (updateSubscriber, messageReceived, messageSent) it can be helpful to show UI notifications

Reply:
Ah yes you are right 🙏